### PR TITLE
Wait to subscribe until after checking for MM

### PR DESCRIPTION
**Pull request by DonDuala**

### CODE REVIEW

File: src/router.ts [hunk 1]
```diff
@@ -180,6 +180,7 @@ class Router {
       for (const dipDeposit of Object.values(this.dips)) {
         if (dipDeposit.qtyTokens > 0) {
           openPositionCount++;
+          // TODO await for a response back from the API
```
Comment (DonDuala) [inline src/router.ts:183]: Any time we route an order successfully, we dupe the run_risk_manager. 1. From CheckMM prices & 2. From position change callback. The position change lags sufficiently where it is picked up after we've awaited on routing.

Either we need something to turn off the callback temporarily or some other fix with separate routing logic for re-runs vs position change. Or whatever you think is a good close solution

**Comment by DonDuala** [comment]
Pretty stuck here. This is the best behavior I've been able to control for. This will handle startup cleanly & a single dip deposit. This will still run multiple RMs whose hedges can collide if there a multiple successive quick deposits. Still this is better than whats currently run and could be merged if we don't want to entirely rework routing immediately

**Comment by DonDuala** [comment]
Could try to create an unsubscribe Poller function to be run during any checkMM

**Comment by brittcyr** [comment]
> Could try to create an unsubscribe Poller function to be run during any checkMM

Is there a problem with routing, or is the issue that we are running too many scalpers? The scalper should be changed to be threadsafe by adding locking. If done correctly, would that solve most of the issues? Would also have to hold the lock during sending to MM, but seems like that would fix it.

**Comment by DonDuala** [comment]
> > Could try to create an unsubscribe Poller function to be run during any checkMM
> 
> Is there a problem with routing, or is the issue that we are running too many scalpers? The scalper should be changed to be threadsafe by adding locking. If done correctly, would that solve most of the issues? Would also have to hold the lock during sending to MM, but seems like that would fix it.

I'm viewing multiple scalpers running as a symptom of incorrect routing. We should only route when we've confirmed there is a net position change, but the poller triggers a callback on any AccountChangeCallback. So in the event the protocol buys and then sells immediately to MM, routing tries to run RM twice. We can either stop the poller during those checkMM periods or do make it threadsafe. Not sure where to begin on the threadsafe portion if you have a reference

**Comment by DonDuala** [comment]
@brittcyr This is cleanly handling DIP deposits at startup & on the fly. Worth running

**Comment by brittcyr** [comment]
> @brittcyr This is cleanly handling DIP deposits at startup & on the fly. Worth running

Is it possible to strip out the unrelated stuff and just make this change be for handling routing?

**Comment by DonDuala** [comment]
> > @brittcyr This is cleanly handling DIP deposits at startup & on the fly. Worth running
> 
> Is it possible to strip out the unrelated stuff and just make this change be for handling routing?

Ya not sure why the merge did that.. Let me fiddle with it

File: src/main.ts [hunk 2]
```diff
@@ -40,9 +44,13 @@ async function main() {
         if (productStatus.get(symbol)) {
           console.log('------------------------------------------------');
           console.log(`RERUN ${symbol} Risk Manager ${new Date().toUTCString()}`);
-          await router.refresh_dips();
+          const [pollers, mmAccounts] = await router.refresh_dips();
```
Comment (brittcyr) [inline src/main.ts:47]: change name of refresh_dips to reflect the return type.

**Review by brittcyr — APPROVED** [review]

**Pull request merged**